entity-service/src/main/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationRepository.kt

```diff
@@ -10,33 +9,31 @@ class Neo4jAuthorizationRepository(
     private val session: Session
 ) {
 
-    data class AvailableRightsForEntity(
-        val targetEntityId: String,
-        val rights: List<String> = emptyList()
-    )
-
-    fun getAvailableRightsForEntities(userId: String, entitiesId: List<String>): List<AvailableRightsForEntity> {
+    fun filterEntitiesUserHasOneOfGivenRights(
+        userId: String,
+        entitiesId: List<String>,
+        rights: Set<String>
+    ): List<String> {
         val query = """
             MATCH (userEntity:Entity {id : ${'$'}userId}), (entity:Entity)
             WHERE entity.id IN ${'$'}entitiesId
             OPTIONAL MATCH (userEntity)-[:HAS_OBJECT]->(right:Attribute:Relationship)-[]->(entity:Entity)
             OPTIONAL MATCH (userEntity)-[:HAS_OBJECT]->(:Attribute:Relationship)-[:IS_MEMBER_OF]->(:Entity)-[:HAS_OBJECT]->(grpRight:Attribute:Relationship)-[]->(entity:Entity)
-            RETURN entity.id as id, right, grpRight
+            WHERE size([label IN labels(right) WHERE label IN ${'$'}rights | 1]) > 0 AND
```

Comment:
I don't get the:  | 1
What it's used for ?

Reply:
This should be removed.